### PR TITLE
Allow multiple packages for `uv tool upgrade/uninstall`

### PR DESCRIPTION
**Pull request by blueraft**
## Summary

Resolves https://github.com/astral-sh/uv/issues/6571

## Test Plan

`cargo test`

### CODE REVIEW

File: crates/uv-cli/src/lib.rs [hunk 1]
```diff
@@ -3121,7 +3121,7 @@ pub struct ToolUninstallArgs {
 pub struct ToolUpgradeArgs {
     /// The name of the tool to upgrade.
     #[arg(required = true)]
-    pub name: Option<PackageName>,
+    pub name: Option<Vec<PackageName>>,
```
Comment (charliermarsh) [inline crates/uv-cli/src/lib.rs:3124]: Can these be `Vec<PackageName>`? I think there's no semantic difference between `None` and `Some(vec![])` in this case.

Comment (blueraft) [reply crates/uv-cli/src/lib.rs]: that's true, I've changed it to `Vec<PackageName>`

**Comment by charliermarsh** [comment]
Looks great, thank you!

File: crates/uv-cli/src/lib.rs (hunk 1 above)
Comment (charliermarsh) [reply crates/uv-cli/src/lib.rs]: Can we do the same for uninstall?

Comment (blueraft) [reply crates/uv-cli/src/lib.rs]: Created a follow up PR: https://github.com/astral-sh/uv/pull/7077

**Pull request merged**